core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -406,10 +380,9 @@ class FsHistoryProviderSuite extends SparkFunSuite with BeforeAndAfter with Matc
       SparkListenerApplicationStart("v10Log", None, 2L, "test", None),
       SparkListenerApplicationEnd(4L))
 
+    // Old-style logs are ignored.
```

Comment:
You should remove `writeOldLog` and code related to it. If you really want to make sure old logs are ignored, just create a directory with some empty files in it, not need to make it look like a proper old log.